File: app/src/main/java/eu/alexanderfischer/dvbverspaetungsinfo/testing/UiTestingHelper.java
package eu.alexanderfischer.dvbverspaetungsinfo.testing;

import android.annotation.TargetApi;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.TaskStackBuilder;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.support.v4.app.NotificationCompat;

import java.util.Random;

import eu.alexanderfischer.dvbverspaetungsinfo.MainActivity;
import eu.alexanderfischer.dvbverspaetungsinfo.R;

/**
 * Created by Alexander Fischer.
 * Class for testing UI specific functionalities.
 */
public class UiTestingHelper {

    /**
     * Tests the Notifications.
     * @param context Activity context.
     */
    @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
    public static void testNotification(Context context) {
        Random rand = new Random();
        int mId = rand.nextInt(10000);

        NotificationCompat.Builder mBuilder =
                new NotificationCompat.Builder(context)
                        .setSmallIcon(R.drawable.ic_notification)
                        .setContentTitle("Hinweis")
                        .setContentText("Test")
                        .setAutoCancel(true);

        Intent resultIntent = new Intent(context, MainActivity.class);

        TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
        stackBuilder.addParentStack(MainActivity.class);
        stackBuilder.addNextIntent(resultIntent);

        PendingIntent resultPendingIntent =
                stackBuilder.getPendingIntent(
                        0,
                        PendingIntent.FLAG_UPDATE_CURRENT
                );

        mBuilder.setContentIntent(resultPendingIntent);

        NotificationManager mNotificationManager =
                (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);

        mNotificationManager.notify(mId, mBuilder.build());
    }

}

File: app/src/main/java/eu/alexanderfischer/dvbverspaetungsinfo/testing/UiTestingHelper.kt
package eu.alexanderfischer.dvbverspaetungsinfo.testing

import android.content.Context
import android.util.Log
import eu.alexanderfischer.dvbverspaetungsinfo.models.Delay
import eu.alexanderfischer.dvbverspaetungsinfo.ui.NotificationHelper
import io.realm.RealmList

/**
 * Created by Alexander Fischer.
 * Class for testing UI specific functionalities.
 */
class UiTestingHelper(context: Context) {
    private val TAG = UiTestingHelper::class.java.simpleName

    init {
        Log.e(TAG, "Test notifications. Not for production.")

        val helper = NotificationHelper(context)

        val negativeDelay = testNegativeDelay()
        helper.sendNotification(negativeDelay)

        val positiveDelay = testPositiveDelay()
        helper.sendNotification(positiveDelay)
    }

    private fun testNegativeDelay(): Delay {
        val delay = Delay()
        delay.state = "negativ"

        val lines = RealmList<String>()
        lines.add("4")

        delay.linien = lines

        return delay
    }

    private fun testPositiveDelay(): Delay {
        val delay = Delay()
        delay.state = "positiv"

        val lines = RealmList<String>()
        lines.add("6")

        delay.linien = lines

        return delay
    }

}
